model/map/src/main/java/org/keycloak/models/map/user/MapUserAdapter.java

```diff
@@ -265,7 +265,8 @@ public void leaveGroup(GroupModel group) {
 
     @Override
     public boolean isMemberOf(GroupModel group) {
-        return RoleUtils.isMember(getGroupsStream(), group);
+        Set<String> groups = entity.getGroupsMembership();
+        return groups != null && groups.contains(group.getId());
```

Comment:
`isMemberOf` accounts for hierarchy. This change modifies the semantics, and is unrelated to legacy store. Please fix.